src/peft/tuners/xlora/model.py

```diff
@@ -86,6 +87,70 @@ def convert_layers_to_xlora(
     return (total_swapped, device)
 
 
+def _load_adapter_into_lora_model(
+    lora_model: LoraModel,
+    adapter_name: str,
+    model_id: str,
+    i: int,
+    torch_device: Optional[str] = None,
+    ephemeral_gpu_offload: bool = False,
+    autocast_adapter_dtype: bool = True,
+    subfolder: Optional[str] = None,
+    **kwargs,
+):
+    """
+    All params pertain to the adapter (adapter name, model id, `i` is the adapter number in 0 indexing)
+    """
+    from peft.peft_model import PeftModel
+    from peft.tuners.lora.config import LoraConfig
+    from peft.utils.other import infer_device
+    from peft.utils.save_and_load import load_peft_weights
+
+    hf_hub_download_kwargs, kwargs = PeftModel._split_kwargs(kwargs)
+    if torch_device is None:
+        torch_device = infer_device()
+
+    if adapter_name not in lora_model.peft_config:
+        # load the config
+        lora_peft_config = LoraConfig.from_pretrained(
+            model_id,
+            ephemeral_gpu_offload=ephemeral_gpu_offload,
+            subfolder=subfolder,
+            **hf_hub_download_kwargs,
+        )
+        lora_peft_config.inference_mode = False
+        lora_model.peft_config[adapter_name] = lora_peft_config
+        lora_model.inject_adapter(lora_model.model, adapter_name)
+
+    adapter_weights = load_peft_weights(model_id, device=torch_device, subfolder=subfolder, **hf_hub_download_kwargs)
+    new_adapter_weights = {}
+    # Rework the keys to contain the adapter numbers
+    for old_key in adapter_weights.keys():
+        key: str = old_key
+        # Remove all the prefixes until we have model.<...>
+        while not (key.startswith("model.") and not key.startswith("model.model.")):
+            key = key[key.find(".") + 1 :]
+        # We always want model.model
+        key = "model." + key
+        new_adapter_weights[key] = adapter_weights[old_key]
+
+    # load the weights into the model
+    ignore_mismatched_sizes = kwargs.get("ignore_mismatched_sizes", False)
+    _load_result = set_peft_model_state_dict(
+        lora_model,
+        new_adapter_weights,
+        adapter_name=adapter_name,
+        ignore_mismatched_sizes=ignore_mismatched_sizes,
+    )
+    if len(_load_result.unexpected_keys) > 0:
```

Comment:
No need for leading underscore

```suggestion
    load_result = set_peft_model_state_dict(
        lora_model,
        new_adapter_weights,
        adapter_name=adapter_name,
        ignore_mismatched_sizes=ignore_mismatched_sizes,
    )
    if len(load_result.unexpected_keys) > 0:
```

Reply:
I changes this, thanks!